### PR TITLE
UserSession class [2] - Refactor mod/ files

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp #12021

### CODE REVIEW

File: mod/cal.php [hunk 1]
```diff
@@ -42,7 +41,7 @@
 
 function cal_init(App $a)
 {
-	if (DI::config()->get('system', 'block_public') && !Session::isAuthenticated()) {
+	if (DI::config()->get('system', 'block_public') && !DI::userSession()->isAuthenticatedDI::userSession()->getRemoteUserId()) {
```
Comment (MrPetovan) [inline mod/cal.php]: Uuuuh...
```suggestion
	if (DI::config()->get('system', 'block_public') && !DI::userSession()->isAuthenticated()) {
```

File: mod/display.php [hunk 2]
```diff
@@ -46,14 +45,14 @@ function display_init(App $a)
 		(new Objects(DI::l10n(), DI::baseUrl(), DI::args(), DI::logger(), DI::profiler(), DI::apiResponse(), $_SERVER, ['guid' => DI::args()->getArgv()[1] ?? null]))->run();
 	}
 
-	if (DI::config()->get('system', 'block_public') && !Session::isAuthenticated()) {
+	if (DI::config()->get('system', 'block_public') && !DI::userSession()->isAuthenticatedDI::userSession()->getRemoteUserId()) {
```
Comment (MrPetovan) [inline mod/display.php]: UUUUUH....
```suggestion
	if (DI::config()->get('system', 'block_public') && !DI::userSession()->isAuthenticated()) {
```

File: mod/display.php [hunk 3]
```diff
@@ -62,18 +61,18 @@ function display_init(App $a)
...
+		if (DI::userSession()->getLocalUserId()) {
+			$item = Post::selectFirstForUser(DI::userSession()->getLocalUserId(), $fields, ['guid' => DI::args()->getArgv()[1], 'uid' => DI::userSession()->getLocalUserId()]);
 			if (DBA::isResult($item)) {
 				$nick = $a->getLoggedInUserNickname();
 			}
 		}
 
 		// Is this item private but could be visible to the remove visitor?
-		if (!DBA::isResult($item) && Session::getRemoteUser()) {
+		if (!DBA::isResult($item) && DI::userSession()->getRemoteUserIdDI::userSession()->getRemoteUserId()) {
```
Comment (MrPetovan) [inline mod/display.php]: Alright, back to the drawing board.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mod/display.php (hunk 3 above)
Comment (nupplaphil) [reply mod/display.php]: Yes, I pushed it too early :D

**Comment by nupplaphil** [comment]
@MrPetovan  - now plz have a look :-)

File: mod/cal.php [hunk 4]
```diff
@@ -112,11 +111,11 @@ function cal_content(App $a)
 	$owner_uid = intval($owner['uid']);
 	$nick = $owner['nickname'];
 
-	$contact_id = Session::getRemoteContactID($owner['uid']);
+	$contact_id = DI::userSession()->getRemoteContactID($owner['uid']);
```
Comment (MrPetovan) [inline mod/cal.php:114]: For consistency the method should be renamed to `getRemoteContactId`.
```suggestion
	$contact_id = DI::userSession()->getRemoteContactId($owner['uid']);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mod/cal.php (hunk 4 above)
Comment (nupplaphil) [reply mod/cal.php:114]: Can I rename this method after all PR at once?
because otherwise I think I will loose my overview :)

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**